sys/include/sema.h

```diff
@@ -58,92 +78,76 @@ typedef struct {
  * @param[in] value Initial value for the semaphore.
  *
  * @return  0 on success.
- * @return  -EINVAL, if semaphore is invalid.
```

Comment:
About this you already complained earlier ([here](https://github.com/RIOT-OS/RIOT/pull/6155#discussion_r89315071) and [here](https://github.com/RIOT-OS/RIOT/pull/6155#discussion_r92342656)). The old return codes were restored (except if they made no sense anymore) and we aded a `@pre` tag. In any case this code is only used inside the posix semaphore and lwip and both have been adapted and tested respectively.